mlflow/utils/search_utils.py

```diff
@@ -852,23 +852,31 @@ def _get_sort_key(cls, order_by_list):
             order_by.append(("experiment_id", False))
 
         # https://stackoverflow.com/a/56842689
-        class _Reversor:
-            def __init__(self, obj):
+        class _Sorter:
+            def __init__(self, obj, ascending):
                 self.obj = obj
+                self.ascending = ascending
 
             # Only need < and == are needed for use as a key parameter in the sorted function
             def __eq__(self, other):
                 return other.obj == self.obj
 
             def __lt__(self, other):
-                return other.obj < self.obj
-
-        def _apply_reversor(experiment, key, ascending):
+                if self.obj is None:
+                    return False
+                elif other.obj is None:
+                    return True
```

Comment:
The existing sort logic doesn't handle NoneTypes, but the creation time / last update time attributes are absent on old experiments created with MLflow 1.20 (file store tested). This PR adjusts the logic.